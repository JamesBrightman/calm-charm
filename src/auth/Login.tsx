import { GoogleAuthProvider, signInWithPopup, signInWithEmailAndPassword, EmailAuthProvider } from "firebase/auth";
import { useNavigate } from "react-router";
import { auth } from "../firebase/firebaseInit";
import {IonItem, IonLabel, IonInput} from "@ionic/react";

export const Login = () => {
    const nav = useNavigate();

    const signInWithGoogle = () => {
        const provider = new GoogleAuthProvider();
        signInWithPopup(auth, provider).then((user) => {
            console.log(user)
            nav("/")
        })
      }

      const signInWithEmailPass = () => {
        const provider = new EmailAuthProvider();
        signInWithEmailAndPassword(auth, "email", "password").then((user) => {
            console.log(user)
            nav("/")
        })
      }

    return (<div className="flex flex-col gap-4 items-center justify-center">
          <IonItem>
        <IonLabel className="bg-red-500">Input with placeholder</IonLabel>
        <IonInput placeholder="Enter company name"></IonInput>
      </IonItem>
      <button onClick={signInWithGoogle}>Sign In</button>
      {/* <button onClick={() => {signOut(auth)}}>Sign Out</button> */}
    </div>)
}