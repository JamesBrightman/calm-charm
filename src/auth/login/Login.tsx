import { useState } from "react";
import { CreateAccountPage } from "./CreateAccountPage";
import { LoginAccountPage } from "./LoginAccountPage";

export const Login = () => {
  const [showLoginPage, setShowLoginPage] = useState<boolean>(true);

  return (
    <div className="w-full h-screen flex flex-col bg-blue-500">
      {showLoginPage ? (
        <LoginAccountPage
          toggleCreate={() => {
            setShowLoginPage(false);
          }}
        />
      ) : (
        <CreateAccountPage
          toggleLogin={() => {
            setShowLoginPage(true);
          }}
        />
      )}
    </div>
  );
};
