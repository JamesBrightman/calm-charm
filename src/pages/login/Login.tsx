import React, { useState } from 'react';
import { CreateAccountPage } from '../../auth/pages/CreateAccountPage';
import { LoginAccountPage } from '../../auth/pages/LoginAccountPage';

export const Login = () => {
  const [showLoginPage, setShowLoginPage] = useState<boolean>(true);

  return (
    <div className="w-full h-screen flex flex-col bg-blue-500">
      {showLoginPage ? (
        <LoginAccountPage
          toggleCreate={() => {
            setShowLoginPage(false);
          }}
        />
      ) : (
        <CreateAccountPage
          toggleLogin={() => {
            setShowLoginPage(true);
          }}
        />
      )}
    </div>
  );
};
