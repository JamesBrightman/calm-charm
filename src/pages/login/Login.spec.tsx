import React from 'react';
import { render } from '@testing-library/react';
import { Login } from './Login';
import { BrowserRouter } from 'react-router-dom';

describe('Login Page', () => {
  it('Login page renders', () => {
    const { baseElement } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    expect(baseElement).toBeTruthy();
  });

  it('Login page matches snapshot', () => {
    const { baseElement } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
