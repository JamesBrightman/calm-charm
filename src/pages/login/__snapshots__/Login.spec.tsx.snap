// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Login page matches snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full h-screen flex flex-col bg-blue-500"
    >
      <div
        class="h-screen flex flex-col flex-1 bg-blue-500"
      >
        <p
          class="flex items-center justify-center pt-8 text-4xl font-bold text-white"
        >
          [ Branding ]
        </p>
        <div
          class="w-full h-content mt-auto flex flex-col"
          style="opacity: 0; transform: translateY(500px) translateZ(0);"
        >
          <svg
            class="mb-[-1px]"
            viewBox="0 0 1440 320"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,256L48,229.3C96,203,192,149,288,122.7C384,96,480,96,576,112C672,128,768,160,864,181.3C960,203,1056,213,1152,197.3C1248,181,1344,139,1392,117.3L1440,96L1440,320L1392,320C1344,320,1248,320,1152,320C1056,320,960,320,864,320C768,320,672,320,576,320C480,320,384,320,288,320C192,320,96,320,48,320L0,320Z"
              fill="white"
              fill-opacity="1"
            />
          </svg>
          <div
            class="flex flex-col gap-4 bg-white items-center justify-center"
          >
            <p
              class="text-4xl font-extrabold px-4 bg-white"
            >
              Welcome back.
            </p>
            <div
              class="flex flex-col items-center justify-center"
            >
              <div>
                <form
                  class="flex flex-col gap-2 p-4 w-full items-center justify-center"
                >
                  <div
                    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="outlined-adornment-password"
                    >
                      Email
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd rounded-lg css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="outlined-adornment-password"
                        name="email"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <svg
                          class="w-6 h-6 m-[8px]"
                          fill="none"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Email
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="outlined-adornment-password"
                    >
                      Password
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd rounded-lg  css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="outlined-adornment-password"
                        name="password"
                        type="password"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                      >
                        <button
                          aria-label="toggle password visibility"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="0"
                          type="button"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                            />
                            <path
                              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Password
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <span
                    class="w-full pb-0.5 bg-gray-300 mb-6 mt-2"
                  />
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium grid grid-flow-col grid-cols-[50px_1fr] items-center justify-items-center rounded-xl border-2 px-1 font-semibold bg-blue-600 text-white p-2 w-72 bg-opacity-100 css-h4t5eh-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                    id=":r2:"
                    tabindex="0"
                    type="submit"
                  >
                    <svg
                      class="w-8 h-8"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M11 7L9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                      />
                    </svg>
                    <p
                      class="pr-[50px]"
                    >
                      Sign In
                    </p>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium grid grid-flow-col grid-cols-[50px_1fr] items-center justify-items-center rounded-xl border-2 px-1 w-72 p-1 font-semibold text-gray-700 bg-opacity-100 css-1pkfx9i-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                id=":r3:"
                tabindex="0"
                type="button"
              >
                <svg
                  class="w-8 h-8"
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 48 48"
                  width="1em"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#FFC107"
                  />
                  <path
                    d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                    fill="#FF3D00"
                  />
                  <path
                    d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                    fill="#4CAF50"
                  />
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#1976D2"
                  />
                </svg>
                <p
                  class="pr-[50px]"
                >
                  Sign in with Google
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="flex flex-col gap-1 bg-stone-200 p-2 rounded-lg text-gray-700 undefined"
            >
              <div
                class="flex flex-row gap-1"
              >
                <p>
                  Don't have an account?
                </p>
                <p
                  class="underline font-bold"
                >
                  Sign up here.
                </p>
              </div>
            </div>
            <p
              class=""
            >
              Legal text TODO
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
